x-pack/platform/plugins/shared/alerting_v2/server/config.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { TypeOf } from '@kbn/config-schema';
+
+export const configSchema = schema.object({
+  esqlRules: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+  }),
+});
+
+export type AlertingV2Config = TypeOf<typeof configSchema>;
```

Comment:
```suggestion
export type PluginConfig = TypeOf<typeof configSchema>;
```

nit: I think, as we are already in the world of the `alerting_v2` plugin, there is no need to be part of our variables explicitly.

Reply:
Renamed in 73f3d421ee780addcefa67d550337943dc6b0f84